Exported configuration file is missing few required TS attributes.

### Expected behavior

The app should run without any error if started with exported configuration file is started in the Typescript. 

### Actual behavior

Currently TS can't compile the script with following error. 

```
 Unable to compile TypeScript:
test_app.ts:24:64 - error TS2345: Argument of type '{ ipc: { enabled: boolean; }; version: string; networkVersion: string; label: string; genesisConfig: { blockTime: number; communityIdentifier: string; maxPayloadLength: number; bftThreshold: number; ... 6 more ...; delegateListRoundOffset: number; }; logger: { ...; }; network: { ...; }; forging: { ...; }; plugins: {...' is not assignable to parameter of type 'Partial<ApplicationConfig>'.
  Types of property 'logger' are incompatible.
    Property 'logFileName' is missing in type '{ fileLogLevel: string; consoleLogLevel: string; }' but required in type '{ logFileName: string; fileLogLevel: string; consoleLogLevel: string; }'.
```

### Steps to reproduce

Use following Typescript script, put it in root of the repo as `test_app.ts` and run with `npx ts-node test_app.ts`

```typescript 
import {
	Application,
	genesisBlockDevnet,
	configDevnet,
} from './sdk';

const app = Application.defaultApplication(genesisBlockDevnet, configDevnet);

app
	.run()
	.then(() => console.info('App is running'))
	.catch(console.error);

```

### Which version(s) does this affect? (Environment, OS, etc...)
5.0.0

--- a/framework/src/application.ts
+++ b/framework/src/application.ts
@@ -93,6 +93,9 @@ const registerProcessHooks = (app: Application): void => {
 };
 
 type InstantiableBaseModule = new (genesisConfig: GenesisConfig) => BaseModule;
+type RecursivePartial<T> = {
+	[P in keyof T]?: RecursivePartial<T[P]>;
+};
 
 export class Application {
 	public config: ApplicationConfig;
@@ -112,7 +115,7 @@ export class Application {
 
 	public constructor(
 		genesisBlock: Record<string, unknown>,
-		config: Partial<ApplicationConfig> = {},
+		config: RecursivePartial<ApplicationConfig> = {},
 	) {
 		// Don't change the object parameters provided
 		this._genesisBlock = genesisBlock;
@@ -146,7 +149,7 @@ export class Application {
 
 	public static defaultApplication(
 		genesisBlock: Record<string, unknown>,
-		config: Partial<ApplicationConfig> = {},
+		config: RecursivePartial<ApplicationConfig> = {},
 	): Application {
 		const application = new Application(genesisBlock, config);
 		application._registerModule(TokenModule);

